src/workflows-app/SubmissionHistory.test.js

```diff
@@ -0,0 +1,561 @@
+import { mockAbortResponse, mockAzureApps, mockAzureWorkspace } from 'src/workflows-app/utils/mock-responses';
+
+// Necessary to mock the AJAX module.
+jest.mock('src/libs/ajax');
+jest.mock('src/libs/notifications.js');
+jest.mock('src/libs/ajax/leonardo/Apps');
+jest.mock('src/libs/nav', () => ({
+  getCurrentUrl: jest.fn().mockReturnValue(new URL('https://app.terra.bio')),
+  getLink: jest.fn(),
+  goToPath: jest.fn(),
+}));
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger');
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn(),
+  };
+});
+jest.mock('src/libs/state', () => ({
+  ...jest.requireActual('src/libs/state'),
+  getUser: jest.fn(),
+}));
+// Mocking feature preview setup
+jest.mock('src/libs/feature-previews', () => ({
+  ...jest.requireActual('src/libs/feature-previews'),
+  isFeaturePreviewEnabled: jest.fn(),
+}));
+jest.mock('src/libs/config', () => ({
+  ...jest.requireActual('src/libs/config'),
+  getConfig: jest.fn().mockReturnValue({}),
+}));
+
+// SubmissionHistory component uses AutoSizer to determine the right size for table to be displayed. As a result we need to
+// mock out the height and width so that when AutoSizer asks for the width and height of "browser" it can use the mocked
+// values and render the component properly. Without this the tests will be break.
+// (see https://github.com/bvaughn/react-virtualized/issues/493 and https://stackoverflow.com/a/62214834)
+const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight');
+const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth');
+
+const runSetData = {
+  run_sets: [
+    {
+      error_count: 0,
+      submission_timestamp: '2022-01-01T12:00:00.000+00:00',
+      last_modified_timestamp: '2022-01-02T13:01:01.000+00:00',
+      record_type: 'FOO',
+      run_count: 1,
+      run_set_id: 'ea001565-1cd6-4e43-b446-932ac1918081',
+      state: 'COMPLETE',
+    },
+    {
+      error_count: 1,
+      submission_timestamp: '2021-07-10T12:00:00.000+00:00',
+      last_modified_timestamp: '2021-08-11T13:01:01.000+00:00',
+      record_type: 'FOO',
+      run_count: 2,
+      run_set_id: 'b7234aae-6f43-405e-bb3a-71f924e09825',
+      state: 'ERROR',
+    },
+  ],
+};
+
+beforeAll(() => {
+  Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 });
+  Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 });
+});
+
+const cbasUrlRoot = 'https://lz-abc/terra-app-abc/cbas';
+const cromwellUrlRoot = 'https://lz-abc/terra-app-abc/cromwell';
+
+beforeEach(() => {
+  getConfig.mockReturnValue({ cbasUrlRoot, cromwellUrlRoot });
+  MenuTrigger.mockImplementation(({ content }) => {
+    return div({ role: 'menu' }, [content]);
+  });
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
```

Comment:
Clearing mocks is done automatically via Jest configuration and doesn't need to be repeated per test.